extensions/emmet/src/extension.ts

```diff
@@ -137,8 +128,32 @@ export function activate(context: vscode.ExtensionContext) {
 			updateExtensionsPath(newExtensionsPath);
 			extensionsPath = newExtensionsPath;
 		}
+		registerCompletionProviders(context);
 	}));
 }
 
+/**
+ * Holds any registered completion providers by their language strings
+ */
+const registeredCompletionProviders: string[] = [];
+
+function registerCompletionProviders(context: vscode.ExtensionContext) {
+	let completionProvider = new DefaultCompletionItemProvider();
+	Object.keys(LANGUAGE_MODES).forEach(language => {
+		const provider = vscode.languages.registerCompletionItemProvider(language, completionProvider, ...LANGUAGE_MODES[language]);
```

Comment:
we should check if the languages here are already registered here as well. Or skip this whole section, if `registerCompletionProviders` is being called for config change